### PR TITLE
Use linting to detect undesired function usage

### PR DESCRIPTION
**Pull request by hadley**

### CODE REVIEW

**Review by jayhesselberth — APPROVED** [review]
Looks good to me, nice that github links the result at the specific line number.

File: .lintr.R [hunk 1]
```diff
@@ -0,0 +1,40 @@
+linters <- list(lintr::undesirable_function_linter(
+  fun = c(
+    # Base messaging
+    "message" = "use cli::cli_inform()",
```
Comment (maelle) [inline .lintr.R:4]: this is so nice!!

**Review by maelle — COMMENTED** [review]
lintr is really great :gem:

**Comment by hadley** [comment]
I just wish I could figure out why that test is now failing. Has it just been silently failing in the past because `dir()` would just return `character()`?

File: tests/testthat/test-templates.R [hunk 2]
```diff
@@ -74,9 +74,11 @@ test_that("find templates in local pkgdown first", {
 
 test_that("BS5 templates have main + aside", {
   names <- dir(path_pkgdown("bs5", "templates"), pattern = "content-")
```
Comment (maelle) [inline tests/testthat/test-templates.R:76]:
```suggestion
  names <- dir(path_pkgdown("BS5", "templates"), pattern = "content-")
```

Comment (hadley) [reply tests/testthat/test-templates.R:76]: Oooooooooooooooooooooh

Comment (hadley) [reply tests/testthat/test-templates.R]: You have no idea how much time this has saved me 🤣 also a good reason why using fs functions are safer than base.

Comment (maelle) [reply tests/testthat/test-templates.R]: so base R functions are not case-sensitive?!

Comment (hadley) [reply tests/testthat/test-templates.R]: No, it's just that `dir()` returns `character()` when the directory doesn't exist; `dir_ls()` errors.

**Pull request merged**